spec/models/piece_spec.rb

```diff
@@ -225,44 +225,39 @@
       expect(king.y_position).to eq 3
     end
 
-    # it 'should move the piece to the destination square; if the square is occupied, mark that piece as captured' do
-    #   game = Game.create(white_user_id: 0, black_user_id: 1)
-    #   king = King.create(x_position: 4, y_position: 2, game_id: game.id)
-    #   rook = Rook.create(x_position: 4, y_position: 3, game_id: game.id)
-    #   king.move_to!(4,3)
-    #   expect(king.x_position).to eq 4
-    #   expect(king.y_position).to eq 3
-    #   expect(rook.captured).to eq true
-    # end
-  end
-
-  describe 'white?' do
-    it 'should return true for white piece' do
-      game = Game.create(white_user_id: 0, black_user_id: 1)
-      pawn = Pawn.create(x_position: 0, y_position: 6, game_id: game.id, user_id: 0)
-      expect(pawn.white?).to be true
+    # test is failing - true is expected, returns nil (the default value)
+    it 'should move the piece to the destination square; if the square is occupied, mark that piece as captured' do
+      white_user = FactoryGirl.create(:user)
+      black_user = FactoryGirl.create(:user)
+      game = Game.create(white_user_id: white_user.id, black_user_id: black_user.id)
+      king = King.create(x_position: 4, y_position: 2, game_id: game.id, user_id: white_user.id)
+      rook = Rook.create(x_position: 4, y_position: 3, game_id: game.id, user_id: black_user.id)
+      king.move_to!(4,3)
+      expect(king.x_position).to eq 4
+      expect(king.y_position).to eq 3
+      expect(rook.captured).to eq true
```

Comment:
This will return nil because rook is a reference to a model instance that fetched its data from the database when it was created.  Then when the king is moved and the rook captured, the database is updated, but that doesn't automatically update the rook model (since it still has the old data).  Models have a nice convenience method called `reload` that reloads their data from the db: http://apidock.com/rails/ActiveRecord/Persistence/reload.

So after `king.move_to` you'll probably want `rook.reload`